### DISCUSSION TITLE
Running "mikro-orm migration:check" without a database connection

### DISCUSSION THREAD

**Discussion by elliegoldstein** [mikro-orm/mikro-orm#5030] [Q&A] [answered]
Hi, I'm trying to run a check in CI to see if there are any migrations that need to be generated before some code is merged. However, in my CI environment I don't have access to a postgres instance, and I get this issue when running `yarn mikro-orm migration:check`

```
mikro-orm migration:check

Check if migrations are needed. Useful for bash scripts.

Options:
  -v, --version  Show version number                                   [boolean]
  -h, --help     Show help                                             [boolean]

Error: No database specified, please fill in `dbName` or `clientUrl` option
```

Is there any way to set up a config without a connection to a database, and check if migration files are up-to-date?

**Comment by B4nan** [comment, marked as answer]
Do you have [a snapshot file](https://mikro-orm.io/docs/next/migrations#snapshots) generated (and committed)? It should be preferred over the actual schema inference.

**Reply by elliegoldstein** [reply]
Yup, I have an up-to-date snapshot file in my migrations folder.

**Reply by elliegoldstein** [reply]
This is what my config looks like, if it helps:

```typescript
import { Options, defineConfig } from "@mikro-orm/postgresql";
const CI_DATABASE_CONFIG: Options = defineConfig({
  metadataProvider: TsMorphMetadataProvider,
  entities: ["./lib/entities"],
  migrations: {
    emit: "ts",
    pathTs: "src/migrations",
    path: "lib/migrations",
  },
});
```

**Reply by elliegoldstein** [reply]
@B4nan is there any change I need to make to my config to use the snapshot file when the database isn't available?

**Reply by elliegoldstein** [reply]
Turns out I needed to add `connect: false` and specify the snapshot DB name in `migrations`

**Reply by petergledhillinclusive** [reply]
We have the same problem since v6.  The `connect: false` makes the `check` command work without trying to connect to a db but it does mean we have to have a different config for this command.

**Reply by B4nan** [reply]
This should help: ef5d14ad0f0ccf153c373991e2545851d10566cf

Nowadays it's actually fine to just disable the eagerly created connection since there are fallbacks that will create one if it's not there yet lazily.
